lib/susedistribution.pm

```diff
@@ -744,7 +744,7 @@ sub activate_console {
         $user ||= 'root';
         handle_password_prompt;
         ensure_user($user);
-        assert_screen(["text-logged-in-$user", "text-login"], 60);
+        assert_screen(["text-logged-in-$user", "tty1-selected"], 60);
```

Comment:
The pseudo-console 'ssh' can be one of **root-console, install-shell, user-console, log-console** and those have different tty.

```perl
# lib/susedistribution.pm
# ...
458         $self->add_console('install-shell',  'tty-console', {tty => 2});                                                                                                                                                                  
459         $self->add_console('installation',   'tty-console', {tty => check_var('VIDEOMODE', 'text') ? 1 : 7});                                                                                                                             
460         $self->add_console('install-shell2', 'tty-console', {tty => 9});                                                                                                                                                                  
461         # On SLE15 X is running on tty2 see bsc#1054782                                                                                                                                                                                   
462         $self->add_console('root-console',   'tty-console', {tty => get_root_console_tty});                                                                                                                                               
463         $self->add_console('user-console',   'tty-console', {tty => 4});                                                                                                                                                                  
464         $self->add_console('log-console',    'tty-console', {tty => 5});                                                                                                                                                                  
465         $self->add_console('displaymanager', 'tty-console', {tty => 7});                                                                                                                                                                  
466         $self->add_console('x11',            'tty-console', {tty => get_x11_console_tty});                                                                                                                                                
467         $self->add_console('tunnel-console', 'tty-console', {tty => 3}) if get_var('TUNNELED');
# ...
737     $type = 'ssh' if (get_var('BACKEND', '') =~ /generalhw/ && !defined(get_var('GENERAL_HW_VNC_IP')) && $console =~ /root-console|install-shell|user-console|log-console/);
```

The tty number has to be known at this place, and then use

```suggestion
        assert_screen(["text-logged-in-$user", "tty$tty-selected"], 60);
```